### DISCUSSION TITLE
Arrows Navigation and Bad Performance with React-Select.

### DISCUSSION THREAD

**Discussion by JanderSilv** [petyosi/react-virtuoso#871] [Q&A]
Thanks for this awesome package @petyosi.

I'm not getting how to make keyboard arrow navigation work with react-select component.

Other point is, using like 3.000 items with React-Select as well, causes a bad performance of the list scroll. So I don't have sure if the list is really being virtualized.

My Attempt:
https://stackblitz.com/edit/react-ts-ypm48f?file=App.tsx

I found a "solution" at React-Select repository, but only works partially:
https://stackblitz.com/edit/react-ts-bbvmhv?file=MenuList.tsx

Behavior:

https://user-images.githubusercontent.com/42092038/228975361-25a8144b-caa6-411f-94cc-ae059a86aa19.mp4

**Comment by petyosi** [comment]
I can't say for sure what's the right way to integrate virtuoso in react-select, not familiar with the component. However, the current implementation receives the *full list of 3000 react nodes* on render - and then on each keystroke. This is further slowed down by the call of `Children.toArray` done in itemContent. 

<img width="736" alt="image" src="https://user-images.githubusercontent.com/13347/229023800-07ab395f-d5c3-420a-8264-9a05fe19592d.png">
